testsuite/tests/libratest/smoke_test.rs

```diff
@@ -1312,3 +1330,190 @@ fn test_key_manager_consensus_rotation() {
+    let script = transaction_builder::encode_set_validator_config_script(
+        account_context.account,
+        consensus_key.to_bytes().to_vec(),
+        validator_network_key.to_bytes(),
+        validator_network_address.into(),
+        full_node_network_key.to_bytes(),
+        full_node_network_address.into(),
+    );
+    submit_new_transaction(libra, time_service, account_context, script)
+}
+
+fn submit_new_reconfig<L: LibraInterface, T: TimeService>(
+    libra: &L,
+    time_service: &T,
+    account_context: &AccountContext,
+) -> anyhow::Result<(AccountAddress, u64)> {
+    let script = Script::new(
+        libra_transaction_scripts::RECONFIGURE_TXN.clone(),
+        vec![],
+        vec![],
+    );
+    submit_new_transaction(libra, time_service, account_context, script)
+}
+
+fn get_libra_interface(node_config: &NodeConfig) -> JsonRpcLibraInterface {
+    let json_rpc_endpoint = format!("http://127.0.0.1:{}", node_config.rpc.address.port());
+    JsonRpcLibraInterface::new(json_rpc_endpoint)
+}
+
+#[test]
+/// There are three main steps to network key rotation
+/// 1. Rotate the key in local storage
+/// 2. Write a transaction to update the ValidatorConfig (Operator)
+/// 3. Make a Reconfigure call (Association)
+fn test_network_key_rotation() {
+    let mut swarm = TestEnvironment::new(5);
+    swarm.validator_swarm.launch();
+
+    // Get association info for reconfig
+    let key = swarm.faucet_key.0;
+    let association_info = AccountContext {
+        public_key: key.public_key(),
+        private_key: key,
+        account: association_address(),
+    };
+
+    // Load the node's config
+    let (node_config_path, node_config) = {
+        let node_config_path = swarm.validator_swarm.config.config_files.get(0).unwrap();
+        (
+            node_config_path.clone(),
+            NodeConfig::load(&node_config_path).unwrap(),
+        )
+    };
+
+    // Load validator's on disk storage
+    let mut storage: Storage = if let Identity::FromStorage(storage_identity) =
+        &node_config.validator_network.as_ref().unwrap().identity
+    {
+        (&storage_identity.backend).try_into().unwrap()
+    } else {
+        panic!("Couldn't load identity from storage");
+    };
+
+    // Get Operator info
+    let operator_key = storage.export_private_key(OPERATOR_KEY).unwrap();
+    let operator_info = AccountContext {
+        account: libra_types::account_address::from_public_key(&operator_key.public_key()),
+        public_key: operator_key.public_key(),
+        private_key: operator_key,
+    };
+
+    // Setup an RPC endpoint so we can talk to the node
+    let validator_account = node_config.validator_network.as_ref().unwrap().peer_id();
+    let libra = get_libra_interface(&node_config);
+
+    // Rotate key in storage
+    storage.rotate_key(VALIDATOR_NETWORK_KEY).unwrap();
+    let new_network_key = { node_config.validator_network.unwrap().identity_key() };
```

Comment:
nit: do you need these braces?

Reply:
I can check if I can remove them.  I spent a lot of time fighting with mutable vs immutable references, and some of these are necessary.